### PR TITLE
chore: remove apex-tmlanguage from the dependencies

### PR DESCRIPTION
**Pull request by mingxuanzhangsfdx**
### What does this PR do?
remove apex-tmlanguage from the dependencies of bundle bc it is never used.

### What issues does this PR fix or reference?
[skip-validate-pr]

### Functionality Before
apex-tmlanguage is added to apex and soql extension

### Functionality After
apex-tmlanguage no longer exists with no effect.

### CODE REVIEW

**Review by peternhale — COMMENTED** [review]
@mingxuanzhangsfdx why is this being removed?

**Comment by mingxuanzhangsfdx** [comment]
@peternhale it is removed because the module `apex-tmlanguage` is not referenced in any code. And the only reason why we introduce the dependency is for the artifacts in the folder `grammars` in `apex` and `soql`. the artifacts in `grammars` are copied during the `[postcompile](https://github.com/forcedotcom/salesforcedx-vscode/blob/mz/remove-apex-tmlanguage/package.json#L74)` phase. And when we generate the `extension` folder for packaging, the grammars folder is already [included](https://github.com/forcedotcom/salesforcedx-vscode/blob/mz/remove-apex-tmlanguage/packages/salesforcedx-vscode-soql/package.json#L134). 

So I do not see a necessity of making it in external.

File: packages/salesforcedx-vscode-apex/package.json [hunk 1]
```diff
@@ -104,7 +104,6 @@
       "main": "dist/index.js",
       "languageServerDir": "dist",
       "dependencies": {
-        "@salesforce/apex-tmlanguage": "1.4.0",
```
[the remark this answers is not in the document]
Comment (gbockus-sf) [reply packages/salesforcedx-vscode-apex/package.json:107]: Also looks like .gitignore is not committing the grammars dir.  So if we need those grammars to be present (which I assume we do)  We'll need to keep this in place

**Review by gbockus-sf — DISMISSED** [review]
I think we have to keep this so we have the grammars 🤷

**Review by peternhale — APPROVED** [review]

**Pull request merged**